packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -0,0 +1,262 @@
+        .auth(token)
+        .request((err, resp, body) => {
+          if (err || resp.statusCode !== 200) {
+            err = this._error(err, resp)
+            logger.error(err, 'provider.zoom.list.error')
+            return done(err)
+          } else {
+            done(null, this._adaptData(body))
+          }
+        })
+    }
+  }
+
+  download ({ id, token, query }, done) {
+    // meeting id + file id required
+    const meetingId = id
+    const fileId = query.recordingId
+    const GET_MEETING_FILES = `/meetings/${meetingId}/recordings`
+
+    const downloadUrlPromise = new Promise((resolve) => {
+      this.client
+        .get(`${BASE_URL}${GET_MEETING_FILES}`)
+        .auth(token)
+        .request((err, resp) => {
+          if (err) {
+            this._downloadError(resp, done)
+            return
+          }
+          // timeline files don't have an ID
+          const file = resp
+            .body
+            .recording_files
+            .find(file => fileId === file.id || fileId === file.file_type)
+          if (!file || !file.download_url) {
+            return this._downloadError(resp, done)
+          }
+          resolve(file.download_url)
+        })
+    })
+    downloadUrlPromise.then((downloadUrl) => {
+      this.client
+        .get(`${downloadUrl}?access_token=${token}`)
+        .request()
+        .on('response', (resp) => {
+          if (resp.statusCode !== 200) {
+            done(this._error(null, resp))
+          } else {
+            resp.on('data', (chunk) => done(null, chunk))
+          }
+        })
+        .on('end', () => {
+          done(null, null)
+        })
+        .on('error', (err) => {
+          logger.error(err, 'provider.zoom.download.error')
+          done(err)
+        })
+    })
+  }
+
+  size ({ id, token, query }, done) {
+    const meetingId = id
+    const fileId = query.recordingId
+    const GET_MEETING_FILES = `/meetings/${meetingId}/recordings`
+
+    return this.client
+      .get(`${BASE_URL}${GET_MEETING_FILES}`)
+      .auth(token)
+      .request((err, resp, body) => {
+        if (err || resp.statusCode !== 200) {
+          return this._downloadError(resp, done)
+        }
+        const file = resp
+          .body
+          .recording_files
+          .find(file => file.id === fileId || file.file_type === fileId)
+
+        if (!file) {
+          return this._downloadError(resp, done)
+        }
+        // timeline files don't have file size, but are typically small json files, should be much less than 1MB
+        const maxExportFileSize = 1024 * 1024
+        done(null, file.file_size || maxExportFileSize)
+      })
+  }
+
+  _initializeData (initialEnd = null) {
+    let end = initialEnd || moment()
+    let start = end.clone().date(1)
+
+    const defaultLimit = end.clone().subtract('months', DEFAULT_RANGE_MOS)
+    const limit = defaultLimit > OLDEST_RECORD ? defaultLimit : OLDEST_RECORD
+
+    const data = {
+      items: []
+    }
```

Comment:
i took out the api call to users to avoid needing another permission (which would return the user email). We need the API call anyways to retrieve the account creation date, so this is added!